views/appView.js

```diff
@@ -0,0 +1,36 @@
+var AppView = Backbone.View.extend({
+  el: $('.container'),
+
+  initialize: function () {
+    this.model.get('videos').fetch({reset: true});
+    
+    this.listenTo(this.model.get('videos'), 'reset', this.renderPage)
+    this.listenTo(this.model, 'change:mainVideo', this.renderMainVideo)
+  },
+
+  events: {
+    'click .search': 'search',
+  },
+
+  search: function () {
+    var searchQuery = $('#search-query').val();
+    this.model.get('videos').searchYoutube(searchQuery);
+  },
+
+  renderPage: function () {
+    var videos = this.model.get('videos')
+    this.model.set('mainVideo', videos.models[0]);
```

Comment:
Setting the main video is fine here (though it's probably better that you do this from the collection or the app model or create function on that app model that would set it) but, renderMainVideo should be calling in response to `mainVideo` changing - like use a listener for it.

That way, it re-renders any time the `mainVideo` changes.